File: image200904/sample/fibo.c
/*BINFMTC:
  fibonacci with simple recursion algorithm.
 */

#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int fibonacci(int i)
{
  if (i < 2) return i;
  return fibonacci(i - 1) + fibonacci(i - 2);
}

int main(int argc, char **argv)
{
  printf("%i\n", fibonacci(atoi(argv[1])));
  return 0;
}

File: 2009/image200904/sample/fibo.ml
(* fibonacci with simple recursion algorithm.
Copyright 2009 Junichi Uekawa
 *)

let rec fibonacci = function
    0 -> 0
  | 1 -> 1
  | v -> fibonacci (v - 1) + fibonacci (v - 2)

let () = print_int (fibonacci 40); print_newline ()

